experimental/packages/opentelemetry-sdk-metrics-base/src/state/MetricStorageRegistry.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { MetricStorage } from './MetricStorage';
+import { isDescriptorCompatibleWith } from '../InstrumentDescriptor';
+import * as api from '@opentelemetry/api';
+import { Maybe } from '../utils';
+import { getConflictResolutionRecipe, getIncompatibilityDetails } from '../view/RegistrationConflicts';
+
+/**
+ * Internal class for storing {@link MetricStorage}
+ */
+export class MetricStorageRegistry {
+  private readonly _metricStorageRegistry = new Map<string, MetricStorage[]>();
+
+  getStorages(): MetricStorage[] {
+    let storages: MetricStorage[] = [];
+    for (const metricStorages of Array.from(this._metricStorageRegistry.values())) {
```

Comment:
```suggestion
    for (const metricStorages of this._metricStorageRegistry.values()) {
```

We don't need to convert iterators to array when using for...of.

Reply:
Good catch, thanks!
Fixed in [2488a4d](https://github.com/open-telemetry/opentelemetry-js/pull/2734/commits/2488a4d2ed441fe4dc6cc6248613c3eb8f1a9d41)